lib/global_reporter_demon.rb

```diff
@@ -18,6 +18,13 @@ def suppress_stderr
   def after_fork
     STDERR.puts "#{Time.now}: Starting Prometheus global reporter pid: #{Process.pid}"
     t = DiscoursePrometheus::Reporter::Global.start($prometheus_client)
+
+    trap('INT')  { t.kill }
```

Comment:
I worry about killing the thread here since the global collector can be making connections to the db and redos which may not be terminated cleanly when the thread is killed. Maybe we can add some sort of flag in the thread itself for it to terminate cleanly?

Reply:
Good idea 👍 . Implemented in https://github.com/discourse/discourse-prometheus/pull/14/commits/7afc28f5da285e8e4b0fac732c405267aff6a59b